AutomatedTesting/Gem/PythonTests/assetpipeline/ap_fixtures/debug_parse_and_compare.py

```diff
@@ -0,0 +1,236 @@
+        assert (data_value in expected_hashes_to_skip), "Scene mismatch"
+    return expected_hashes_to_skip, actual_hashes_to_skip
+
+
+def scan_scene_debug_scene_graph_file_differences_for_issues(diff_actual: List[str], diff_expected: List[str],
+                                                             actual_file_path: str,
+                                                             expected_file_path: str) -> (List[str], List[str]):
+    """
+    GHI #13714
+    Hashes are generated differently on different platforms right now, short term, this is a known issue,
+    Long term, it will be important to stabilize the output of product assets, so this will emit a warning,
+    to help keep track of the related GHI.
+    diff_actual: A list containing the debug output (dbgsg file) of the processed source asset under test.
+    diff_expected: A list containing the expected debug output of the source asset under test.
+    actual_file_path: Path to the dbgsg file containing the debug output of the processed source asset under test.
+    expected_file_path: Path to the dbgsg file containing the expected debug output to compare against.
+    """
+
+    diff_actual_hashes_removed = []
+    diff_expected_hashes_removed = []
+
+    hash_regex = re.compile("(.*Hash: )([0-9]*)")
+
+    actual_hashes = []
+    expected_hashes = []
+
+    for list_entry in diff_actual:
+        match_result = hash_regex.match(list_entry)
+        assert match_result, "Scene mismatch"
+        diff_actual_hashes_removed.append(match_result.group(1))
+        actual_hashes.append(match_result.group(2))
+
+    for list_entry in diff_expected:
+        match_result = hash_regex.match(list_entry)
+        assert match_result, "Scene mismatch"
+        diff_expected_hashes_removed.append(match_result.group(1))
+        expected_hashes.append(match_result.group(2))
+
+    hashes_removed_diffs_identical = utils.compare_lists(diff_actual_hashes_removed, diff_expected_hashes_removed)
+
+    # If, after removing all of the hash values, the lists are now identical, emit a warning.
+    if hashes_removed_diffs_identical == True:
+        warnings.warn(
+            f"Hash values no longer match for debug scene graph between files {expected_file_path} and {actual_file_path}")
+
+    return expected_hashes, actual_hashes
+
+
+def compare_scene_debug_file(asset_processor: object, expected_file_path: str, actual_file_path: str,
+                             expected_hashes_to_skip: List[str] = None,
+                             actual_hashes_to_skip: List[str] = None) -> tuple[None, None] | tuple[list[str], list[str]]:
+    """
+    GHI #13713 and GHI #13714
+    Loads both the debug scene graph generated by re-processing the test scene file and the last known good debug scene
+    graph for that file. Scans both files for differences, emitting warnings on known issues and failing on
+    unknown issues.
```

Comment:
If you can follow the pattern in the other docstring updates here, this will be perfect.